optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolverManager.java

```diff
@@ -148,17 +149,16 @@ public SolverStatus getSolverStatus(ProblemId_ problemId) {
     //        solverJob.reloadProblem(problemFinder);
     //    }
 
-    // TODO Future features
-    //    @Override
-    //    public void addProblemFactChange(ProblemId_ problemId, ProblemFactChange<Solution_> problemFactChange) {
-    //        DefaultSolverJob<Solution_, ProblemId_> solverJob = problemIdToSolverJobMap.get(problemId);
-    //        if (solverJob == null) {
-    //            // We cannot distinguish between "already terminated" and "never solved" without causing a memory leak.
-    //            logger.debug("Ignoring addProblemFactChange() call because problemId ({}) is not solving.", problemId);
-    //            return;
-    //        }
-    //        solverJob.addProblemFactChange(problemFactChange);
-    //    }
+    @Override
+    public void addProblemChange(ProblemId_ problemId, ProblemChange<Solution_> problemChange) {
+        DefaultSolverJob<Solution_, ProblemId_> solverJob = getSolverJob(problemId);
+        if (solverJob == null) {
+            // We cannot distinguish between "already terminated" and "never solved" without causing a memory leak.
+            throw new IllegalStateException("Cannot add the problem change because there is no solver solving the problemId ({"
```

Comment:
```suggestion
            throw new IllegalStateException("Cannot add the problem change (" + problemChange + ") because there is no solver solving the problemId ("
```